kb/development/deprecation.md

```diff
@@ -1,22 +1,25 @@
 # Deprecations
 
-To ensure backwards-compatibility between versions of Mbed TLS, functions in the public interface may not be removed except during the release of a new major version. Instead, functions may be 'deprecated', indicating to users that they will be removed in the future.
+To ensure backwards-compatibility between versions of Mbed TLS, functions or constants in the public interface may not be removed except during the release of a new major version. Instead, they may be 'deprecated', indicating to users that they will be removed in the future.
+
+When a build is made with `MBEDTLS_DEPRECATED_WARNING` defined, a compiler warning will be generated to warn a user that the function or constant will be removed at some point in the future, notifying users that they should change to a replacement function or constant at their own convenience. To find examples of past deprecations, look at the previous major version of Mbed TLS.
```

Comment:
It may be too much information here, but `MBEDTLS_DEPRECATED_WARNING` only works with GCC-like compilers. You get a build error if the compiler doesn't recognize `__attribute__((deprecated))`.